extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -888,6 +888,10 @@ String generateSequenceName(
   @VisibleForTesting
   String generateSequenceName(int groupId)
   {
+    if (taskGroups.get(groupId) == null) {
```

Comment:
It looks that this should never happen. Would you elaborate more on when this can happen?

Reply:
This happens if no events were passed to kafka tasks, this is the issue from https://github.com/druid-io/druid/issues/5666. So `checkTaskDuration()` removes the groupId `taskGroups.remove(groupId);` but later `checkPendingCompletionTasks()` tries to get the groupId in `sequenceTaskGroup.remove(generateSequenceName(groupId));`

Reply:
@jihoonson I tried again, removing the null check, and I cannot reproduce the `NPE` now with the task resume fix in, but I think, this null check couldn't hurt.

Reply:
@surekhasaharan thanks. Null check is always good, but I'm not sure about returning null in this method. If `groupId` is never expected to be null, we should throw an exception. Otherwise, this method can return null, but all callers should check the returned sequenceName is null or not. What do you think?

Reply:
I agree with @jihoonson -- let's fix this in a different patch, since it's really a different bug from the main one we're fixing. The main one being:

> If the endOffset is same as startOffset, still let the task resume instead of returning
> endOffsets early which causes the tasks to pause forever and ultimately fail on timeout

I raised a new issue for this NPE: #5900

Reply:
@jihoonson Agree that all the caller will need to handle null in this case. Will address the possible NPE in separate issue raised by @gianm . Removing the null check here.